File: frontend/src/api/types/playlist.ts
export type Playlist = {
  id: number
  title: string
  description: string | null
  song_ids: number[] | null
  user_ids: number[] | null
  created_at: string
  updated_at: string
}

export type PlaylistCreate = {
  title: string
  description?: string | null
  song_ids?: number[]
}

export type PlaylistUpdate = {
  title?: string | null
  description?: string | null
  song_ids?: number[] | null
}

File: backend/app/schemas/playlist.py
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel


class PlaylistBase(BaseModel):
    title: str
    description: Optional[str] = None


class PlaylistCreate(BaseModel):
    title: str
    description: Optional[str] = None
    song_ids: List[int] = []


class PlaylistUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    song_ids: Optional[List[int]] = None


class PlaylistRead(BaseModel):
    id: int
    title: str
    description: Optional[str] = None
    song_ids: Optional[List[int]] = None
    user_ids: Optional[List[int]] = None

    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

